source/extensions/filters/listener/proxy_protocol/proxy_protocol.cc

```diff
@@ -443,6 +456,24 @@ ReadOrParseState Filter::readProxyHeader(Network::ListenerFilterBuffer& buffer)
   } else {
     // continue searching buffer from where we left off
     for (; search_index_ < raw_slice.len_; search_index_++) {
+
+      if (config_.get()->allowRequestsWithoutProxyProtocol()) {
+        if (search_index_ < PROXY_PROTO_V1_SIGNATURE_LEN &&
+            buf[search_index_] != PROXY_PROTO_V1_SIGNATURE[search_index_]) {
+          possibly_v1_ = false;
```

Comment:
seems `possibly_v1_` and `possibly_v2` needn't to be the class member, since you only care each byte is equal or not.

Reply:
that's not true; removing them would mean we could no longer tell if we had a signature that was partially valid v1 proxy protocol and partially valid v2 proxy protocol

this test covers that case (and would fail if we removed `possibly_v1_` and `possibly_v2_` class members): https://github.com/envoyproxy/envoy/blob/85706e635a4d5c0a4d4ef8ba008b3c431245d95b/test/extensions/filters/listener/proxy_protocol/proxy_protocol_test.cc#L254-L273

Reply:
made this optimization in a slightly different way in https://github.com/envoyproxy/envoy/pull/18951/commits/c86edef7feb35cd3d268f73c1020ba2b713c6d93